### PR TITLE
actor: refactor and add unit tests

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Long Dai <long0dai@foxmail.com>

# Description

Add unit tests for actor.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by daixiang0** [comment]
please help review @artursouza

**Comment by artursouza** [comment]
> please help review @artursouza


The E2E tests and build are not passing. The change does a refactoring too, not only adding tests. Please, make sure the E2E tests are passing before we can review.

File: pkg/actors/actors.go [hunk 1]
```diff
@@ -879,62 +878,72 @@ func (a *actorsRuntime) CreateTimer(ctx context.Context, req *CreateTimerRequest
...
 
-		err := a.executeTimer(req.ActorType, req.ActorID, req.Name, req.DueTime,
-			req.Period, req.Callback, req.Data)
-		if err != nil {
-			log.Errorf("error invoking timer on actor %s: %s", actorKey, err)
-		}
+	err := a.executeTimer(req.ActorType, req.ActorID, req.Name, req.DueTime,
+		req.Period, req.Callback, req.Data)
+	if err != nil {
+		log.Debugf("error invoking timer on actor %s: %s", actorKey, err)
```
Comment (artursouza) [inline pkg/actors/actors.go:920]: Why change this verbosity?

File: pkg/actors/actors.go [hunk 2]
```diff
@@ -879,62 +878,72 @@ func (a *actorsRuntime) CreateTimer(ctx context.Context, req *CreateTimerRequest
...
+
+	for {
+		select {
+		case <-ticker.C:
+			_, exists := a.actorsTable.Load(actorKey)
+			if exists {
+				err := a.executeTimer(req.ActorType, req.ActorID, req.Name, req.DueTime,
+					req.Period, req.Callback, req.Data)
+				if err != nil {
+					log.Debugf("error invoking timer on actor %s: %s", actorKey, err)
```
Comment (artursouza) [inline pkg/actors/actors.go:933]: Why change in verbosity?

**Review by artursouza — CHANGES_REQUESTED** [review]

File: pkg/actors/actors.go (hunk 1 above)
Comment (daixiang0) [reply pkg/actors/actors.go:920]: Wrong changes, revert.

**Pull request closed without merging**